docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -685,6 +685,130 @@ since legacy applications don't call these API functions explicitly.
+| 4. `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT + flag` | `oe_attestation_header \|\| oe_report_header \|\| SGX_ECDSA_quote(custom_claims)` |
+| 5. `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` | `SGX_ECDSA_quote(custom_claims)` |
+| 6. `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE + flag` | `oe_attestation_header \|\| SGX_ECDSA_quote(custom_claims)` |
+
+For all evidence structures above, the hearts of the evidences are the same - SGX ECDSA_P256 quote. SGX ECDSA_P256 quote should to be like any other evidence format which is presented in 2 formats - with or without embedded format id. For example, format XYZ is presented as:
+1. Without flag - `Evidence_XYZ`
+2. With flag - `oe_attestation_header || Evidence_XYZ || custom_claims`
+
+That is, we only need to support these evidence structures:
+- Format 2: Generated by previous version oe_get_evidence().
+- Format 3: Generated by legacy oe_get_report() which will eventually be deprecated.
+- Format 5: Generated by SGX SDK.
+
+In the proposal, flag `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` indicates wraping the plugin evidence with a header containing format ID, and hashing the custom claims and passing the hash to the plugin.
+
+Consider renaming the flag to reflect the fact.
+
+### User Experience
+
+#### Attester
+- To generate format 2:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID,
+    custom_claims,
+    size_t custom_claims_size,
+    ...);
+```
+
+- To generate format 3:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT,
+    0,  // Must be 0
+    custom_claims,
+    size_t custom_claims_size,  // Must be <= 64 bytes
+    ...);
+```
+
+- To generate format 5:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    0,
+    custom_claims, 
+    size_t custom_claims_size,  // Must be <= 64 bytes
+    ...);
+```
+
+#### Verifier
+- To verify format 2:
+```
+oe_verify_evidence(
+    NULL,
+    ...);
+```
+
+- To verify format 3:
+```
+oe_verify_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT,
+    ...);
+```
+
+- To verify format 5:
+```
+oe_verify_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    ...);
+```
+
+### Plan
+Reduction/deprecation is planned in phases.
+
+Here 'Remove' means the API definition is removed from public header files. Referring to the removed API definition results in compilation error.
+'Deprecate' means the API definition is redefined in public header files. Referring to the deprecated API definition does not break the build but the behavior is changed internally to guarantee continuous support.
+
+#### Phase 1 - V0.11
+- Remove `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` as proposed above.
+
+#### Phase 2 - V0.11
+- Remove `oe_report_header` from format 2. Increment `oe_attestation_header.version` to flag the change.
+- Modify `oe_verify_evidence()`, `oe_verify_report()`, and `oe_verify_attestation_certificate()` so they handle new format version.
+- Deprecate `oe_verify_attestation_certificate_with_evidence`.
```

Comment:
Can you elaborate on this deprecation and eventual removal? Nothing else in this PR seems like it explains the reasoning for this or its eventual removal. Given that this API was just added, it seems weird that we're already deprecating it. Or did you mean the older `oe_verify_attestation_certificate` API?

Similarly, should this also be deprecating the `oe_generate_attestation_certficate` at the same time?

Reply:
oe_verify_attestation_certificate_with_evidence() was added to differentiate the x.509 extension being an oe_report or oe_evidence. Since the OIDs are different, we did not need this additional API at the first place.

Reply:
Thanks for the explanation, could you add that to the document for clarity?

Also, is there a plan for the existing versions of those functions that don't call into `oe_get_evidence`/`oe_verify_evidence` in the deprecation plan?